### ISSUE TITLE
PiranhaJava // Remove meaningless "assert" and "expect" from JUnits

### ISSUE THREAD

**Issue by pranavsb** [uber/piranha#47]
Piranha simplification will need manual review and fixing of testcases, but I think it's useful to delete lines like:
`expect(true).andReturn(true);`
`assertFalse(true);`
which Piranha leaves behind in test classes after simplifying toggle expressions.

**Comment by pranavsb** [comment]
What's the best way to approach this?

I got it working by trying some basic checks in 
`public Description matchMethod(MethodTree tree, VisitorState state)` which is [here](https://github.com/uber/piranha/blob/39a150940454a26dbf60a2ed0406690c63610dc9/java/piranha/src/main/java/com/uber/piranha/XPFlagCleaner.java#L718)


Very hacky code:
```
AnnotationTree tat =
            ASTHelpers.getAnnotationWithSimpleName(tree.getModifiers().getAnnotations(), "Test");
    if (tat != null) {
      // easymock - most common is andReturn - add more to list but have to be 100% sure
      if (tree.getBody().toString().contains("expect(true).andReturn") || tree.getBody().toString().contains("expect(false).andReturn")) {
        Description.Builder builder = buildDescription(tree);
        SuggestedFix.Builder fixBuilder = SuggestedFix.builder();
        List<? extends StatementTree> statements = tree.getBody().getStatements();
        for (StatementTree statement : statements) {
          if (isSimplifiedTestStatement(statement.toString())) {
            fixBuilder.delete(statement);
            builder.addFix(fixBuilder.build());
            return builder.build();
//            decrementAllSymbolUsages(tree, state, fixBuilder);
          }
        }
}

```

where `isSimplifiedTestStatement` just does more string matching. 

Not sure if adding this logic to `matchMethod` would work, maybe we need to run piranha twice or modify the code to increase no. of passes.

**Comment by mkr-plse** [comment]
Can you post the input code and expected output code? I think updating the `methodProperties` for these methods should handle this issue too.

**Comment by pranavsb** [comment]
Basically, delete the lines which are test related calls, where Piranha has already visited and substituted with a boolean

Original code

```
@Test
    public void testToggle() {
        ...
        int a = 1;
        expect(mockToggle.isToggleEnabled()).andReturn(true).anyTimes();
        int b = a + 1;
        ...
        assertFalse(mockToggle.isToggleEnabled());
    }
```

Piranha modified code

```
@Test
    public void testToggle() {
        ...
        int a = 1;
        expect(true).andReturn(true).anyTimes();
        int b = a + 1;
        ...
        assertFalse(true); // causes test to fail
    }
```

Expected code:

```
@Test
    public void testToggle() {
        ...
        int a = 1;
        int b = a + 1;
        ...
    }
```

**Comment by mkr-plse** [comment]
This may be a slightly involved fix and it is better to avoid special casing. We will need to implement multi-pass analysis on the code to get this correct. 

1. In the first pass, determine the various Piranha refactorings and their locations, but do no rewrite the code. 
2. In the second pass, use the information from the previous pass along with standard Piranha refactoring to rewrite. 

For the second pass to succeed in the case given above, methods `expect`, `andReturn` and `anyTimes` can be given as `empty` methodTypes as part of `methodProperties`. 

Another option is to determine the code range that is updated by Piranha and re-run the standard Piranha analysis and restrict the refactorings to those regions. 

Let us discuss more on the gitter channel before finalizing a solution.

**Comment by ketkarameya** [comment]
This has been resolved in #156.
